### PR TITLE
[Merged by Bors] - feat(Algebra/GroupPower): Miscellaneous lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
Generalise `pow_ite`/`ite_pow` and give a version of `pow_add_pow_le` that doesn't require the exponent to be nonzero.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Basic.lean [hunk 1]
```diff
@@ -1055,3 +1056,27 @@ theorem multiplicative_of_isTotal (p : α → Prop) (hswap : ∀ {a b}, p a →
...
+@[to_additive (attr := simp) ite_smul]
+lemma pow_ite (p : Prop) [Decidable p] (a : α) (b c : β) :
+    a ^ (if p then b else c) = if p then a ^ b else a ^ c := pow_dite _ _ _ _
+
+@[to_additive (attr := simp) smul_ite]
+lemma ite_pow (p : Prop) [Decidable p] (a b : α) (c : β) :
+    (if p then a else b) ^ c = if p then a ^ c else b ^ c := dite_pow _ _ _ _
+
+set_option linter.existingAttributeWarning false in
+attribute [to_additive (attr := simp)] dite_smul smul_dite ite_smul smul_ite
```
Comment (jcommelin) [inline Mathlib/Algebra/Group/Basic.lean:1080]: Why are these needed? Is this a known issue? If so, please mention it in a comment above these lines. If not, please post the bug on zulip.

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by YaelDillies** [comment]
bors merge
